src/screens/NetworkSettings/NetworkSettingsFlowStack.js

```diff
@@ -1,15 +1,27 @@
 import { createStackNavigator } from '@react-navigation/stack';
+import { useSelector } from 'react-redux';
+import { eq } from 'lodash';
+import { PRE_SETTINGS_MAINNET, PRE_SETTINGS_TESTNET } from '../../constants';
 import { CustomNetworkSettingsNav, CustomNetworkSettingsScreen } from './CustomNetworkSettingsScreen';
 import { NetworkPreSettingsNav, NetworkPreSettingsScreen } from './NetworkPreSettingsScreen';
 import { NetworkSettingsDislaimerNav, NetworkSettingsDislaimerScreen } from './NetworkSettingsDisclaimerScreen';
 
 export const NetworkSettingsFlowNav = Symbol('NetworkSettingsFlowStack').toString();
 
+const isCustomNetwork = (networkSettings) => {
+  return !(eq(PRE_SETTINGS_MAINNET, networkSettings) || eq(PRE_SETTINGS_TESTNET, networkSettings));
+};
+
 export const NetworkSettingsFlowStack = ({ navigation }) => {
+  const networkSettings = useSelector((state) => state.networkSettings);
+  const initialNav = isCustomNetwork(networkSettings)
+    ? NetworkPreSettingsNav
+    : NetworkSettingsDislaimerNav;
+
   const FlowStack = createStackNavigator();
   return (
     <FlowStack.Navigator
-      initialRouteName={CustomNetworkSettingsNav}
+      initialRouteName={initialNav}
```

Comment:
We also need a warning and confirmation when changing to the testnet. Do we have it?

Reply:
I didn't understood. Can you clarify? Let me tell what I understood:
- You are saying we need to show the "disclaimer" screen even if the user is using the "testnet" network?
- You are saying we need to take a confirmation when the user pick the "testnet" pre-settings?

Reply:
Oh! Ok. I got it now. You are saying we need to show the disclaimer screen for all the users, not matter the networkSettings state. Sure, this is a simpler solution.

Reply:
- Make disclaimer screen required for any network state: 225c9b53d904051e7a2e23505e4b1364a09b0ce5